### PR TITLE
Don't allow a Swagger property description to be null

### PR DESCRIPTION
**Pull request by jhollinger**
Currently, if you're using response documentation and your Grape entity omits `desc` from `documentation`:

    class Entities::Widget < Grape::Entity
      expose :id, documentation: {type: 'integer'} # some properties are just super obvious
      ...
    end

that property is rendered to this JSON:

    "properties": {
      "id": {
        "type": "integer",
        "description": null
      },
      ...
    }

and Swagger UI chokes on it with:

    Uncaught TypeError: Swagger 2.0 does not support null types ([object Object]).  See https://github.com/swagger-api/swagger-spec/issues/229.

IMHO the `description` field should be omitted in these cases.

### CODE REVIEW

**Comment by jhollinger** [comment]
I've rebased this branch against master, so it's clean to merge again. It still appears to be an issue.

File: lib/ruby-swagger/grape/type.rb [hunk 1]
```diff
@@ -102,7 +102,7 @@ def type_convert(type, with_definition = true)
               end
             end
 
-            cursor['properties'][target]['description'] = definition[:documentation][:desc] if definition[:documentation].present?
+            cursor['properties'][target]['description'] = definition[:documentation][:desc] if definition[:documentation].present? and definition[:documentation][:desc].present?
```
Comment (rngtng) [inline lib/ruby-swagger/grape/type.rb]: what about:
```ruby
cursor['properties'][target]['description'] = definition[:documentation][:desc] if definition.fetch(:documentation, {})[:desc].present?
```

later we could use some ruby 2.3 foo too ;)

**Comment by rngtng** [comment]
any chance to write a spec for this too? otherwise :+1:

**Comment by jhollinger** [comment]
Sure thing. I tweaked the `Status` test entity and it's tests to fail if this change gets reverted.

**Comment by oliverzeyen** [comment]
#46 fixed this

**Comment by oliverzeyen** [comment]
Can be closed since #46 refactors type.rb completely and fixes this issue.

**Pull request closed without merging**